src/main/java/catan/services/util/play/PlayUtil.java

```diff
@@ -0,0 +1,135 @@
+                    if (gameUser.getMoveOrder() == game.getCurrentMove()) {
+                        actionCode = getActionCodeInPreparationStage(game, initialBuildingsSet);
+                    }
+                    if (!actionCode.isEmpty()) {
+                        actionsList.add("{\"code\": \"" + actionCode.get(0).toString() + "\", \"params\": {}}");
+                        gameUser.setAvailableActions("{\"list\": [" + actionsList.get(0) + "], \"isMandatory\": true}");
+                    } else {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                    }
+                    actionCode.clear();
+                    actionsList.clear();
+                }
+                break;
+            case MAIN:
+                //TODO: complete this part when developing main stage part
+                for (GameUserBean gameUser : game.getGameUsers()) {
+                    if (gameUser.getMoveOrder() == game.getCurrentMove()) {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                    } else {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                        //non-active users have not any available actions (trade in future)
+                    }
+                }
+                break;
+            default:
+                log.debug("Cannot recognize current game stage: {}", game.getStage());
+                throw new GameException(ERROR_CODE_ERROR);
+        }
+    }
+
+    private List<GameUserActions> getActionCodeInPreparationStage (GameBean game, List<List<String>> initialBuildingsSet) {
+        List<GameUserActions> actionCode = new ArrayList<GameUserActions>();
+        if (isUserBuiltAllBuildingsInCurrentCycle(game, initialBuildingsSet)) {
+            actionCode.add(GameUserActions.END_TURN);
+            //set end turn mandatory
+        } else if (getCurrentInitialBuilding(game, initialBuildingsSet).equals("ROAD")) {
+            actionCode.add(GameUserActions.BUILD_ROAD);
+            //set build road mandatory
+        } else if (getCurrentInitialBuilding(game, initialBuildingsSet).equals("SETTLEMENT")) {
+            actionCode.add(GameUserActions.BUILD_SETTLEMENT);
+            //set build settlement mandatory
+        } else if (getCurrentInitialBuilding(game, initialBuildingsSet).equals("CITY")) {
+            actionCode.add(GameUserActions.BUILD_CITY);
+            //set build city mandatory
+        }
+        return actionCode;
+    }
+
+    private boolean isUserBuiltAllBuildingsInCurrentCycle(GameBean game, List<List<String>> initialBuildingsSet) {
+        return game.getCurrentCycleBuildingNumber() > initialBuildingsSet.get(game.getPreparationCycle() - 1).size();
+    }
+
+    private String getCurrentInitialBuilding(GameBean game, List<List<String>> initialBuildingsSet) {
```

Comment:
maybe rename all methods related only to PREPARATION stage to something like ...InPreparationStage? or even move them to PreparationStageUtil? but it is not significant, just if you want and have enough time